src/thread.ts

```diff
@@ -140,226 +146,200 @@ export class Thread<SCG extends ExtendableGenerics = DefaultGenerics> {
       parentMessage,
       participants,
       createdAt,
+      deletedAt,
       updatedAt,
     } = thread.state.getLatestValue();
 
-    this.state.next((current) => {
-      const pendingReplies = Array.from(this.pendingRepliesMap.values());
+    // Preserve pending replies and append them to the updated list of replies
+    const pendingReplies = Array.from(this.failedRepliesMap.values());
 
-      return {
-        ...current,
-        read,
-        replyCount,
-        replies: replies.length ? replies.concat(pendingReplies) : replies,
-        parentMessage,
-        participants,
-        createdAt,
-        updatedAt,
-        isStateStale: false,
-      };
+    this.state.partialNext({
+      read,
+      replyCount,
+      replies: pendingReplies.length ? replies.concat(pendingReplies) : replies,
+      parentMessage,
+      participants,
+      createdAt,
+      deletedAt,
+      updatedAt,
+      isStateStale: false,
     });
   };
 
-  /**
-   * Makes Thread instance listen to events and adjust its state accordingly.
-   */
-  // eslint-disable-next-line sonarjs/cognitive-complexity
   public registerSubscriptions = () => {
-    // check whether this instance has subscriptions and is already listening for changes
-    if (this.unsubscribeFunctions.size) return;
-
-    // TODO: figure out why markAsRead needs to be wrapped like this (for tests to pass)
-    const throttledMarkAsRead = throttle(() => this.markAsRead(), DEFAULT_MARK_AS_READ_THROTTLE_DURATION, {
-      leading: true,
-      trailing: true,
-    });
+    if (this.unsubscribeFunctions.size) {
+      // Thread is already listening for events and changes
+      return;
+    }
 
-    this.unsubscribeFunctions.add(
-      this.state.subscribeWithSelector(
-        (nextValue) => [nextValue.active, this.client.userID && nextValue.read[this.client.userID]?.unreadMessageCount],
-        ([active, unreadMessagesCount]) => {
-          if (!active || !unreadMessagesCount) return;
+    this.unsubscribeFunctions.add(this.subscribeMarkActiveThreadRead());
+    this.unsubscribeFunctions.add(this.subscribeReloadActiveStaleThread());
+    this.unsubscribeFunctions.add(this.subscribeMarkThreadStale());
+    this.unsubscribeFunctions.add(this.subscribeNewReplies());
+    this.unsubscribeFunctions.add(this.subscribeRepliesRead());
+    this.unsubscribeFunctions.add(this.subscribeReplyDeleted());
+    this.unsubscribeFunctions.add(this.subscribeMessageUpdated());
+  };
 
-          // mark thread as read whenever thread becomes active or is already active and unread messages count increases
-          throttledMarkAsRead();
-        },
-      ),
+  private subscribeMarkActiveThreadRead = () =>
+    this.state.subscribeWithSelector(
+      (nextValue) => [nextValue.active, ownUnreadCountSelector(this.client.userID)(nextValue)],
+      ([active, unreadMessageCount]) => {
+        if (!active || !unreadMessageCount) return;
+        this.markAsRead();
```

Comment:
Throttling in this case might be necessary as in a thread that is pretty active (read "getting spammed by other people") this could be a cause for rate-limitting.

Reply:
Good point. I'm going to denounce the underlying request.

Reply:
Fixed in https://github.com/GetStream/stream-chat-js/commit/805ad1809e5650c1a40059cb90e38ac19ac55641.